line-sdk/src/main/java/com/linecorp/linesdk/openchat/ui/ProfileInfoFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.linecorp.linesdk.openchat.ui
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+
+import com.linecorp.linesdk.R
+import com.linecorp.linesdk.databinding.ProfileInfoFragmentBinding
+import com.linecorp.linesdk.openchat.TextUpdateWatcher
+import kotlinx.android.synthetic.main.profile_info_fragment.displayNameEditText
+import kotlinx.android.synthetic.main.activity_create_open_chat.toolbar
+
+class ProfileInfoFragment : Fragment() {
+
+    private lateinit var binding: ProfileInfoFragmentBinding
+
+    companion object {
+        fun newInstance() = ProfileInfoFragment()
+    }
+
+    private lateinit var viewModel: OpenChatInfoViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = DataBindingUtil.inflate(inflater, R.layout.profile_info_fragment, container, false)
+        binding.lifecycleOwner = this
+
+        return binding.root
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        val activity = activity ?: return
+
+        viewModel = ViewModelProviders.of(activity).get(OpenChatInfoViewModel::class.java)
+        binding.viewModel = viewModel
+
+        setupViews()
+    }
+
+    private fun setupViews() {
+        setupToolbar()
+        setupProfileName()
+    }
+
+    private fun setupProfileName() {
+        displayNameEditText.addTextChangedListener(
+            TextUpdateWatcher(
+                ::updateProfileName,
+                OpenChatInfoViewModel.MAX_CHAT_NAME_LENGTH
+            )
+        )
+
+        displayNameEditText.setText(viewModel.profileName.value)
+    }
+
+    private fun updateProfileName(displayName: String, textLengthString: String) {
+        viewModel.setProfileName(displayName)
+    }
+
+    private fun setupToolbar() {
+        val toolbar = activity?.toolbar ?: return
+
+        toolbar.title = getString(R.string.openchat_create_profile_title)
+        toolbar.menu.clear()
+        toolbar.inflateMenu(R.menu.menu_profile_info)
+        val doneMenuItem = toolbar.menu.findItem(R.id.menu_item_create_profile_done)
+        doneMenuItem.isEnabled = viewModel.isValid.value ?: true
+
+        viewModel.isProfileValid.observe(this, Observer { isValid ->
+            doneMenuItem.isEnabled = isValid ?: true
+        })
+
+        doneMenuItem.setOnMenuItemClickListener {
+            (activity as? CreateOpenChatActivity)?.run { createChatroom() }
```

Comment:
As mentioned above, we can move the create chatroom task into the view model, then we can observe `isChatroomCreated` live data to finish activity if successful.
```kotlin
viewModel.createChatroom()
```